p2p/src/peer.rs

```
@@ -546,7 +546,11 @@ where
                     }
                 }
             };
-            let decoded: Result<T, _> = Decode::decode(&mut data.as_slice());
+            let mut decoded: Result<T, _> = DecodeAll::decode_all(data.as_slice());
+            if decoded.is_err() {
```

can we check for a specific error or error message?

---

Why don't we just agree that buffers must be depleted? I would just return the error even if there are more bytes in the input. This seems like unnecessary complexity

---

We need to deprecate the older approach, and our tests aren't the only thing that could break. I suggested to @Arjentix to leave this behaviour for the time being, so that we can find this in the Java/JavaScript SDKs.

---

I see, can we then open an issue about it?

---

Be my guest.

---

Created #2384